File: python-emd-master/example1.py
from collections import namedtuple
from emd import emd
from math import sqrt

Feature = namedtuple("Feature", ["x", "y", "z"])


def distance(f1, f2):
    return sqrt( (f1.x - f2.x)**2  + (f1.y - f2.y)**2 + (f1.z - f2.z)**2 )


def main():
    features1 = [Feature(100, 40, 22), Feature(211, 20, 2),
                 Feature(32, 190, 150), Feature(2, 100, 100)]
    weights1  = [0.4, 0.3, 0.2, 0.1]
    
    features2 = [Feature(0, 0, 0), Feature(50, 100, 80), Feature(255, 255, 255)]
    weights2  = [0.5, 0.3, 0.2]
    
    print emd( (features1, weights1), (features2, weights2), distance )


if __name__ == "__main__":
    main()

File: python-emd-master/example1.c
/* example1.c */

#include <stdio.h>
#include <math.h>

#include "emd.h"

float dist(feature_t *F1, feature_t *F2)
{
  int dX = F1->X - F2->X, dY = F1->Y - F2->Y, dZ = F1->Z - F2->Z;
  return sqrt(dX*dX + dY*dY + dZ*dZ); 
}

main()
{
  feature_t   f1[4] = { {100,40,22}, {211,20,2}, {32,190,150}, {2,100,100} },
	      f2[3] = { {0,0,0}, {50,100,80}, {255,255,255} };
  float       w1[5] = { 0.4, 0.3, 0.2, 0.1 },
              w2[3] = { 0.5, 0.3, 0.2 };
  signature_t s1 = { 4, f1, w1},
	      s2 = { 3, f2, w2};
  float       e;

  e = emd(&s1, &s2, dist, 0, 0);

  printf("emd=%f\n", e);
}
